src/Collection/CollectionInterface.php

```diff
@@ -915,9 +915,10 @@ public function zipWith($items, $callable);
      * ```
      *
      * @param int $chunkSize The maximum size for each chunk
+     * @param boolean $preserveKeys If the keys of the array should be preserved
      * @return \Cake\Collection\CollectionInterface
      */
-    public function chunk($chunkSize);
+    public function chunk($chunkSize, $preserveKeys = false);
```

Comment:
Unfortunately changing the Interface in the master branch is not permitted. What if we have a `chunkWithKeys()` instead?